### ISSUE TITLE
Perf enhancement for app-header: remove nodes from the template

### ISSUE THREAD

**Issue by frankiefu** [PolymerElements/app-layout#183]
Currently the template in `app-header` contains some nodes that are there for background effects.
```
<template>
  <div id="background">
    <div id="backgroundRearLayer"></div>
    <div id="backgroundFrontLayer"></div>
  </div>
  ...
</template>
```
Those nodes are only needed if you are using an effect that affects background, e.g. `blend-background`, `fade-background` and `parallax-background`.  We should make those nodes not created by default and only creating them when the `effects` property contains an effect that needs those background nodes.  We can have a convention that if the effect is doing background effect and needs those background nodes the effect name needs to end with `-background`.

**Comment by frankiefu** [comment]
Another option is to simply `display: none` on the #background div and only make it `display: block` when background effect is used.

**Comment by blasten** [comment]
I like `display: none` better although it would still require counting b/c the nodes can be used by multiple effects.

**Comment by blasten** [comment]
Fixed in #215
